src/plone/app/multilingual/subscriber.py

```diff
@@ -53,16 +53,15 @@ def reindex_language_independent(ob, event):
             if lif != parent:
                 lif[ob.id].indexObject()
     # Re-index objects deeper inside language independent folder
+    # Force re-index objects inside folders in all languages
     else:
-        language_tool = getToolByName(ob, 'portal_languages')
-        language_codes = language_tool.supported_langs
-        parent_uuid = IUUID(parent).split('-')[0] + '-'
-        for code in language_codes:
-            results = pc.unrestrictedSearchResults(UID=parent_uuid + code)
-            # When we have results, parent has been indexed and we can reindex:
-            for brain in results:
-                tmp = ob.unrestrictedTraverse(brain.getPath() + '/' + ob.id)
-                tmp.reindexObject()
+        brains = pc.unrestrictedSearchResults(portal_type='LIF')
```

Comment:
When I search the catalog by 'LIF' portal_type, the results returns a number of results equal to the languages supported. For this reason I think that it is ok.